serde_v8/src/de.rs

```diff
@@ -339,12 +350,20 @@ impl<'de, 'a, 'b, 's, 'x> de::Deserializer<'de>
 
     // Magic Buffer
     if name == magic::buffer::BUF_NAME {
-      let zero_copy_buf =
-        v8::Local::<v8::ArrayBufferView>::try_from(self.input)
+      let zero_copy_buf = match self.input.is_array_buffer() {
+        // ArrayBuffer
+        true => v8::Local::<v8::ArrayBuffer>::try_from(self.input)
```

Comment:
match-boolean is better to write as if-else

Reply:
As discussed offline my reasoning is that we're functionally mapping a union/enum type where tag is this external check so match seems more fitting as a map. But clippy has a https://rust-lang.github.io/rust-clippy/master/#match_bool pedantic rule, so I'll explore applying that in a follow-up PR